jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -597,6 +612,12 @@ public Object getAttribute(String name)
             if (HttpConnection.class.getName().equals(name) &&
                 _channel.getHttpTransport() instanceof HttpConnection)
                 return _channel.getHttpTransport();
+            if (SpecComplianceListener.class.getName().equals(name))
+            {
+                if (_channel instanceof SpecComplianceListener)
+                    return _channel;
```

Comment:
I think this is the wrong return (a channel, not a listener).   Eitherway, it should just call the `getSpecComplianceListener()` above.